cpp/src/arrow/compute/api_scalar.h

```diff
@@ -583,10 +639,32 @@ Result<Datum> MinElementWise(
 ///
 /// \param[in] arg the value to extract sign from
 /// \param[in] ctx the function execution context, optional
-/// \return the elementwise sign function
+/// \return the element-wise sign function
 ARROW_EXPORT
 Result<Datum> Sign(const Datum& arg, ExecContext* ctx = NULLPTR);
 
+/// \brief Round a value to the given number of digits. Array values can be of
+/// arbitrary length. If argument is null the result will be null.
```

Comment:
Hmm... what does "of arbitrary length" imply here? I'm not sure I understand.

Reply:
This phrase is used several times in the function descriptions, I simply followed others as a "template". It refers to the input array, in that it is not required to have a specified length.